code/modules/power/smes_portable.dm

```diff
@@ -0,0 +1,287 @@
+		max_charge += power_cell.maxcharge
+		new_charge += power_cell.charge
+	capacity = max_charge
+	if(!initial(charge) && !charge)
+		charge = new_charge
+
+/obj/machinery/power/smesbank/update_overlays()
+	. = ..()
+	if(machine_stat & BROKEN)
+		return
+
+	if(panel_open)
+		return
+
+	. += "smes-op[connected_port?.outputting ? 1 : 0]"
+	. += "smes-oc[connected_port?.inputting ? 1 : 0]"
+	var/clevel = chargedisplay()
+	if(clevel > 0)
+		. += "smes-og[clevel]"
+
+/obj/machinery/power/smesbank/proc/chargedisplay()
+	return clamp(round(5.5*charge/capacity),0,5)
+
+/obj/machinery/power/smesbank/default_deconstruction_crowbar(obj/item/crowbar/crowbar)
+	if(istype(crowbar) && connected_port)
+		balloon_alert(usr, "disconnect from [connected_port] first!")
+		return FALSE
+	return ..()
+
+// adapted from portable atmos connection code
+/obj/machinery/power/smesbank/wrench_act(mob/living/user, obj/item/wrench)
+	if(connected_port)
+		wrench.play_tool_sound(src)
+		if(wrench.use_tool(src, user, 8 SECONDS))
+			user.visible_message( \
+				"[user] disconnects [src].", \
+				span_notice("You unfasten [src] from [connected_port]."), \
+				span_hear("You hear a ratchet."))
+			investigate_log("was disconnected from [connected_port] by [key_name(user)].", INVESTIGATE_ENGINE)
+			disconnect_port()
+			update_appearance()
+			return ITEM_INTERACT_SUCCESS
+	var/obj/machinery/power/smes/connector/possible_connector = locate(/obj/machinery/power/smes/connector) in loc
+	if(!possible_connector)
+		to_chat(user, span_notice("There's no power connector to connect to."))
+		return ITEM_INTERACT_BLOCKING
+	wrench.play_tool_sound(src)
+	if(wrench.use_tool(src, user, 4 SECONDS))
+		if(!connect_port(possible_connector))
+			to_chat(user, span_notice("[src] failed to connect to [possible_connector]."))
+			return ITEM_INTERACT_BLOCKING
+		user.visible_message( \
+			"[user] connects [src].", \
+			span_notice("You fasten [src] to [possible_connector]."), \
+			span_hear("You hear a ratchet."))
+		update_appearance()
+		investigate_log("was connected to [possible_connector] by [key_name(user)].", INVESTIGATE_ENGINE)
+		return ITEM_INTERACT_SUCCESS
+
+/// Attempt to connect the portable SMES to a given connector. Adapted from portable atmos connection code.
+/obj/machinery/power/smesbank/proc/connect_port(obj/machinery/power/smes/connector/possible_connector)
+	//Make sure not already connected to something else
+	if(connected_port || !possible_connector || possible_connector.connected_smes || possible_connector.panel_open)
+		return FALSE
+
+	//Make sure are close enough for a valid connection
+	if(possible_connector.loc != get_turf(src))
+		return FALSE
+
+	//Perform the connection
+	connected_port = possible_connector
+	connected_port.connected_smes = src
+	possible_connector.on_connect_smes()
+	set_anchored(TRUE) //Prevent movement
+	pixel_x = possible_connector.pixel_x
+	pixel_y = possible_connector.pixel_y
```

Comment:
Oh yeah, by the by we also have pixel_w and pixel_z, which you should also consider because the map_format is now SIDE_MAP which means pixel_x and pixel_y can make atoms render behind other objects depending on their pixel position, while pixel_w and pixel_z don't.